includes/core-changes/corefx/5.0/orderby-firstordefault-complexity-increase.md

```diff
@@ -0,0 +1,40 @@
+### Complexity of LINQ OrderBy increased
+
+The implementation of <xref:System.Linq.Enumerable.OrderBy%2A> has changed, resulting in an increased complexity.
+
+#### Change description
+
+In .NET Core 1.x - 3.x, `OrderBy().First()` and `OrderBy().FirstOrDefault()` operate with `O(N)` complexity. Since only the first (or default) element is required, only one enumeration is required to find it. However, the supplied predicate is invoked exactly `N` times, where `N` is the length of the sequence.
```

Comment:
This raised a question for me (to the product team). Are the same changes used for the following:

- `OrderBy().Last()`
- `OrderBy().LastOrDefault()`
- `OrderByDescending().First()`
- `OrderByDescending().FirstOrDefault()`
- `OrderByDescending().Last()`
- `OrderByDescending().LastOrDefault()`

Reply:
It applies to OrderByDescending.First() and OrderByDescending.FirstOrDefault() as well, according to https://github.com/dotnet/runtime/issues/31554#issuecomment-630288652. I'll update the article, good catch.